packages/gatsby-plugin-netlify/src/build-headers-program.js

```diff
@@ -26,10 +26,19 @@ function validHeaders(headers) {
     headers,
     (headersList, path) =>
       _.isArray(headersList) &&
-      _.every(
-        headersList,
-        header => _.isString(header) && getHeaderName(header)
-      )
+      _.every(headersList, header => {
+        if (_.isString(header)) {
+          if (!getHeaderName(header)) {
+            reporter.warn(
+              `${path} contains an invalid header (${header}). Please check your plugin configuration`
```

Comment:
In the terminal it doesn't say that this came from `gatsby-plugin-netlify`, so it be hard for users to figure out what's going on - let's mention plugin name in this message

Reply:
oh yeah, let me do that